tests/test_javascript.py

```diff
@@ -23,6 +24,12 @@ def test_wait_for_defined_with_reload(self):
         javascript.reload_and_trigger_output()
         self.assertEquals(javascript.output, "Done")
 
+    def test_wait_for_defined_failure(self):
+        with self.assertRaises(BrokenPromise):
```

Comment:
Without the above changes in javascript.py,  this test raises:
```
WebDriverException: Message: something is not defined
```

With the changes, it will correctly keep trying and timeout, raising a BrokenPromise.